### PR TITLE
Set Up Testing Framework and Unit Tests

### LINKED ISSUE #4: Testing API's
#### Description

As a developer, I want to establish a robust testing infrastructure for the project to ensure its reliability and maintainability. Setting up a testing framework and writing unit tests will enable us to catch bugs early, validate changes, and maintain confidence in the codebase.

#### Tasks

- [x] Set up a testing framework (e.g., Jest)
- [x] Configure Jest for TypeScript
- [x] Create a __tests__ directory for unit tests
- [x] Add a test script to package.json

#### Developer Notes

- Utilize Jest as the testing framework for its simplicity and flexibility
- Configure Jest to work seamlessly with TypeScript for type safety
- Organize unit tests within a designated directory for clarity and organization
- Ensure the test script is properly configured and runnable via npm or yarn

#### Definition of Done

- A testing framework, preferably Jest, is integrated into the project
- Jest is configured to handle TypeScript files for effective testing
- A dedicated __tests__ directory is created to house unit tests
- The package.json file includes a script for running tests, making it convenient for developers to execute tests with a single command

## Libraries and there corresponding types
jest
ts-jest
supertest
@types/jest
@types/supertest

### PR DESCRIPTION
**Pull request by wayneleon1**
### What does this PR do?

This PR establishes a testing framework and incorporates unit tests into the project, enhancing the codebase's reliability and facilitating future development by ensuring that individual components function correctly in isolation.

### Description of Task to be completed?
- Set up a testing framework (e.g., Jest)
- Configure Jest for TypeScript
- Create a tests directory for unit tests
- Add a test script to package.json

### How should this be manually tested?
- Checkout branch `ch-testing-apis-#4 `
- Install packages, and run `npm run test`

### Run Commands
- for Testing : `npm run test`
-  for Test coverage : `npm  run test:ci`

### Any background context you want to provide?
For testing purposes, you have to stop the server ; however, I will address this issue in the future.

### CODE REVIEW

**Review by ambroisegithub — APPROVED** [review]
That's look cool!!!

**Review by EddyShimwa — APPROVED** [review]
For me It Looks Good! 👍

**Review by jkarenzi — APPROVED** [review]

**Comment by jkarenzi** [comment]
Looks good to me!

**Review by 13XAVI — COMMENTED** [review]
your codes Looks Great !

**Review by 13XAVI — APPROVED** [review]

File: src/output.log [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+::1 - - [28/Apr/2024:10:32:32 +0000] "GET /ws HTTP/1.1" 404 141 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36"
+::1 - - [28/Apr/2024:10:33:37 +0000] "GET /ws HTTP/1.1" 404 141 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36"
+::1 - - [28/Apr/2024:10:35:46 +0000] "GET /ws HTTP/1.1" 404 141 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36"
+::1 - - [28/Apr/2024:10:40:03 +0000] "GET /ws HTTP/1.1" 404 141 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36"
+::ffff:127.0.0.1 - - [28/Apr/2024:18:18:30 +0000] "GET / HTTP/1.1" 200 57 "-" "-"
+::1 - - [28/Apr/2024:18:47:13 +0000] "GET /ws HTTP/1.1" 404 141 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36"
```
Comment (Habinezajanvier) [inline src/output.log]: Can you force this file in `gitignore`

**Comment by wayneleon1** [comment]
i have done it

**Review by Habinezajanvier — APPROVED** [review]
LGTM

**Pull request merged**